code/modules/surgery/organs/external/tails.dm

```diff
@@ -52,92 +41,115 @@
 		organ_owner.add_mood_event("tail_lost", /datum/mood_event/tail_lost)
 		organ_owner.add_mood_event("tail_balance_lost", /datum/mood_event/tail_balance_lost)
 
-/obj/item/organ/external/tail/generate_icon_cache()
-	. = ..()
-	if((wag_flags & WAG_WAGGING))
-		. += "wagging"
-	return .
-
-/obj/item/organ/external/tail/get_global_feature_list()
-	return GLOB.tails_list
-
 /obj/item/organ/external/tail/proc/wag(mob/user, start = TRUE, stop_after = 0)
 	if(!(wag_flags & WAG_ABLE))
 		return
 
 	if(start)
-		render_key = "wagging[initial(render_key)]"
-		wag_flags |= WAG_WAGGING
+		start_wag()
 		if(stop_after)
 			addtimer(CALLBACK(src, PROC_REF(wag), FALSE), stop_after, TIMER_STOPPABLE|TIMER_DELETE_ME)
 	else
-		render_key = initial(render_key)
-		wag_flags &= ~WAG_WAGGING
+		stop_wag()
 	owner.update_body_parts()
 
+///We need some special behaviour for accessories, wrapped here so we can easily add more interactions later
+/obj/item/organ/external/tail/proc/start_wag()
+	var/datum/bodypart_overlay/mutant/tail/accessory = bodypart_overlay
+	wag_flags |= WAG_WAGGING
+	accessory.wagging = TRUE
+
+///We need some special behaviour for accessories, wrapped here so we can easily add more interactions later
+/obj/item/organ/external/tail/proc/stop_wag()
+	var/datum/bodypart_overlay/mutant/tail/accessory = bodypart_overlay
+	wag_flags &= ~WAG_WAGGING
+	accessory.wagging = FALSE
+
+///Tail parent type (which is MONKEEEEEEEEEEE by default), with wagging functionality
+/datum/bodypart_overlay/mutant/tail
+	layers = EXTERNAL_FRONT|EXTERNAL_BEHIND
+	feature_key = "tail_monkey"
+	var/wagging = FALSE
+
+/datum/bodypart_overlay/mutant/tail/get_base_icon_state()
+	return (wagging ? "wagging_" : "") + sprite_datum.icon_state //add the wagging tag if we be wagging
```

Comment:
This is why tails aren't working. I changed the formatting of the tails to m_tail_spikes_FRONT and m_waggingtail_spikes_FRONT to make it easier to work with. I also did this with functional wings IIRC, you'll need to rename your tails and wings in the .dmi to follow this formatting

Reply:
Thank you for your help!

Reply:
Okay so uhm I'm a fucking idiot and I wasnt on the right branch. The names are formatted as m_tail_lizard_smooth_BEHIND and m_tail_lizard_wagging_smooth_BEHIND